testcases/OpTestSwitchEndianSyscall.py

```diff
@@ -0,0 +1,173 @@
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+# @package OpTestSwitchEndianSyscall
+#  Switch endian system call package for OpenPower testing.
+#
+#  This class will test the functionality of following.
+#  1. It will test switch_endian() system call by executing the registers
+#     changes in other endian. By calling switch_endian sys call,
+#     should not effect register and memory space.
+#  2. This functionality is implemented in linux git repository
+#     /linux/tools/testing/selftests/powerpc/switch_endian
+#  3. In this test, will clone latest linux git repository and make required
+#     files. At the end will execute switch_endian_test executable file.
+
+import time
+import subprocess
+import re
+
+from common.OpTestBMC import OpTestBMC
+from common.OpTestIPMI import OpTestIPMI
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+from common.OpTestLpar import OpTestLpar
+from common.OpTestUtil import OpTestUtil
+
+
+class OpTestSwitchEndianSyscall():
+    ## Initialize this object
+    #  @param i_bmcIP The IP address of the BMC
+    #  @param i_bmcUser The userid to log into the BMC with
+    #  @param i_bmcPasswd The password of the userid to log into the BMC with
+    #  @param i_bmcUserIpmi The userid to issue the BMC IPMI commands with
+    #  @param i_bmcPasswdIpmi The password of BMC IPMI userid
+    #  @param i_ffdcDir Optional param to indicate where to write FFDC
+    #
+    # "Only required for inband tests" else Default = None
+    # @param i_lparIP The IP address of the LPAR
+    # @param i_lparuser The userid to log into the LPAR
+    # @param i_lparPasswd The password of the userid to log into the LPAR with
+    #
+    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_lparip=None,
+                 i_lparuser=None, i_lparPasswd=None):
+        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
+        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
+                                  i_ffdcDir)
+        self.cv_LPAR = OpTestLpar(i_lparip, i_lparuser, i_lparPasswd)
+        self.util = OpTestUtil()
+
+    ##
+    # @brief  If git command availble on lpar, this function will clone linux
+    #         git repository and check for switch_endian_test directory and make
+    #         the required files. And finally execute bin file switch_endian_test.
+    #
+    # @param
+    #
+    # @return BMC_CONST.FW_SUCCESS-success or BMC_CONST.FW_FAILED-fail
+    #
+    def testSwitchEndianSysCall(self):
+
+        # Get OS level
+        self.cv_LPAR.lpar_get_OS_Level()
+
+        # Check whether git command is available on lpar
+        self.cv_LPAR.lpar_check_command("git")
+
+        # Clone latest linux git repository
+        self.cv_LPAR.lpar_clone_linux_source()
```

Comment:
This interface (lpar_clone_linux_source) should take a parameter for which dir to extract the source.  The later function, check_dir_exists() assumes it's /tmp/linux but now we have a coupling between 2 distinct objects.  If we pass in the dir to extract to, it makes it a more flexible interface and allows this class to know for sure where it's going.